storybookjs/storybook
# Vue: Add Vue 3 support

Issue: #10654

## What I did

This PR adds initial support for Vue 3 through `@storybook/vue3` and CLI detection for Vue 3 with `sb init`.

It is heavily inspired by earlier prototypes by:
* https://github.com/storybookjs/storybook/pull/13384 by @SasanFarrokh
* https://github.com/andoshin11/storybook-vue3 by @andoshin11
* https://github.com/milewski/storybook/tree/vue3 by @milewski
* https://github.com/storybookjs/storybook/issues/12632

__All of whom did the initial exploration to get Vue 3 working in storybook!__

Notably missing: Vue 3 support in addon-docs and addon-storyshots, which will be added as individual features once this is in alpha. 😄 

## How to test

- Is this testable with Jest or Chromatic screenshots? Added an e2e test and generator CLI jest tests
- Does this need a new example in the kitchen sink apps? Added vue-3-cli example
- Does this need an update to the documentation? Probably? Any tips?

If your answer is yes to any of these, please make sure to include it in your PR.
